src/load_torrent.cpp

```diff
@@ -16,9 +16,8 @@ see LICENSE file.
 namespace libtorrent {
 
 namespace {
-	void update_atp(add_torrent_params& atp)
+	void update_atp(std::shared_ptr<torrent_info> ti, add_torrent_params& atp)
```

Comment:
As I understand it, a torrent_info is composed of the immutable info dict part, in addition to the trackers, comment, etc.

Would it be preferable to have a non const torrent info structure, that encapsulates a const info dict structure?

Reply:
the mutable type that contains everything in a torrent file, including things outside of the info dictionary is `add_torrent_params`. `torrent_info` will be transitioned to only contain the immutable information from the info-dictionary.

Reply:
> `torrent_info` will be transitioned to only contain the immutable information from the info-dictionary.

👍

Reply:
I think you're absolutely right. One obstacle right now is that `file_storage` is used both for creating new torrents as well as representing existing (immutable) torrents. It makes that class much more complex and less efficient than it would need to. My plan is to shift away from using `file_storage` when creating new torrents, and instead use a separate class. The new API for creating torrents has been implemented in the `master` branch and documented here: https://github.com/arvidn/libtorrent/blob/master/docs/upgrade_to_2.1.rst#creating-torrents

Perhaps I misunderstand and what you're saying is that using `add_torrent_params` both for resume data and non-info-dict .torrent file fields is problematic. I agree with that the name of that class is not great when used to represent a .torrent file, but I think it's a pretty good fit functionality wise.

Reply:
> Perhaps I misunderstand and what you're saying is that using `add_torrent_params` both for resume data and non-info-dict .torrent file fields is problematic.

I don't remember talking about it... But since we are talking about this, I believe that such reuse does not bring any real benefit, except for the momentary benefit of writing a slightly smaller amount of code. Even if these data coincide in some way, I would still make their difference explicit, i.e. I would have a separate class representing a torrent file (and another one for a magnet link), so that specific functions (e.g. parsing/generating routines)  accept parameters of the corresponding types, and a couple of helpers that would generate `add_torrent_params` from their data and vice versa (you could even store `add_torrent_params` as a private field of that classes, it is still better than re-use it directly).

>I agree with that the name of that class is not great when used to represent a .torrent file, but I think it's a pretty good fit functionality wise.

I would still use a name that reflects what this structure represents, and not what it is used for, especially since now it is used not only to add a torrent, but also to save resume data. IMO, even such a simple name as `torrent_data` is much better.